gen/app-gen

```diff
@@ -212,10 +214,49 @@ if [ ${#images[@]} -lt 1 ]; then
     if [[ "${orchestrator}" == "${DOCKER_COMPOSE_ORCHESTRATOR}" ]]; then
         echo "No specific images specified. Will try to extract from docker-compose.yaml file." >&2
 
-        while read -r img; do
-            images+=("$img")
-            images+=("$img")
-        done < <( docker compose --project-directory "$manifests_dir" config --images)
+        if [[ "$manifests_current_dir" == "$manifests_new_dir" ]]; then
+            while read -r img; do
+                images+=("$img")
+                images+=("$img")
+            done < <( docker compose --project-directory "$manifests_current_dir" config --images)
+        else
+            declare -a current_images=()
+            declare -a new_images=()
+
+            while read -r img; do
+                current_images+=("$img")
+            done < <( docker compose --project-directory "$manifests_current_dir" config --images)
+
+            while read -r img; do
+                new_images+=("$img")
+            done < <( docker compose --project-directory "$manifests_new_dir" config --images)
+
+            for i in "${!current_images[@]}"; do
+                for j in "${!new_images[@]}"; do
+                    # if it's a full match, don't include because it was already included in previous docker-compose
+                    if [ "${current_images[i]}" == "${new_images[j]}" ]; then
+                        unset 'current_images[i]'
+                        unset 'new_images[j]'
+                        continue 2
+                    # if it's a match up until the tag, include for update
+                    elif [ "${current_images[i]%%:*}" == "${new_images[j]%%:*}" ]; then
+                        images+=("${current_images[i]}")
+                        images+=("${new_images[j]}")
+                        unset 'current_images[i]'
+                        unset 'new_images[j]'
+                        continue 2
+                    fi
+                done
+                # if it isn't a match, don't include because it's removed from docker-compose
+                unset 'current_images[i]'
+            done
```

Comment:
nice. have you considered using associative arrays (with `declare -A array`) perhaps it would save you some trouble here? if you decide to keep it this way, I would move the ifs to `[[` instead single bracket.